acinclude/krb5.m4

```diff
@@ -8,125 +8,156 @@
-
+  AS_IF([test "x$squid_cv_broken_krb5_h" = "xyes"],[
+    AC_DEFINE(HAVE_BROKEN_SOLARIS_KRB5_H,1,[Define to 1 if Solaris krb5.h is broken for C++])
+  ])
+])
 
-AC_DEFUN([SQUID_CHECK_KRB5_HEIMDAL_BROKEN_KRB5_H], [
+AC_DEFUN([SQUID_CHECK_KRB5_HEIMDAL_BROKEN_KRB5_H],[
   AC_CACHE_CHECK([for broken Heimdal krb5.h],squid_cv_broken_heimdal_krb5_h, [
     SQUID_STATE_SAVE(squid_krb5_heimdal_test)
     CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
-    AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <krb5.h>
-int
-main(void)
-{
-        krb5_context context;
-
-        krb5_init_context(&context);
-
-        return 0;
-}
-]])], [ squid_cv_broken_heimdal_krb5_h=no ], [
-    AC_RUN_IFELSE([AC_LANG_SOURCE([[
+    AC_RUN_IFELSE([
+      AC_LANG_PROGRAM([[#include <krb5.h>]],[[krb5_context c; krb5_init_context(&c);]])
+    ],[squid_cv_broken_heimdal_krb5_h=no],[
+      AC_RUN_IFELSE([
+        AC_LANG_PROGRAM([[
 #define HAVE_BROKEN_HEIMDAL_KRB5_H  1
 #include "compat/krb5.h"
-int
-main(void)
-{
-        krb5_context context;
+        ]],[[krb5_context c; krb5_init_context(&c);]])
+      ],[squid_cv_broken_heimdal_krb5_h=yes],[squid_cv_broken_heimdal_krb5_h=no])
+    ])
+    SQUID_STATE_ROLLBACK(squid_krb5_heimdal_test)
+  ])
+  AS_IF([test "x$squid_cv_broken_heimdal_krb5_h" = "xyes"],[
+    AC_DEFINE(HAVE_BROKEN_HEIMDAL_KRB5_H,1,[Define to 1 if Heimdal krb5.h is broken for C++])
+  ])
+])
 
-        krb5_init_context(&context);
+dnl checks that krb5 is functional. Sets squid_cv_working_krb5
+AC_DEFUN([SQUID_CHECK_WORKING_KRB5],[
+  AS_CASE("$1",
+    ["Heimdal"],[SQUID_CHECK_KRB5_HEIMDAL_BROKEN_KRB5_H],
+    ["Solaris"],[SQUID_CHECK_KRB5_SOLARIS_BROKEN_KRB5_H]
+  )
+  AC_CACHE_CHECK([for working krb5],squid_cv_working_krb5,[
+    SQUID_STATE_SAVE(squid_krb5_test)
+    CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
+    AC_RUN_IFELSE([
+      AC_LANG_PROGRAM([[#include "compat/krb5.h"]],[[krb5_context c; krb5_init_context(&c);]])
+    ],[squid_cv_working_krb5=yes],[squid_cv_working_krb5=no],[:])])
+  SQUID_STATE_ROLLBACK(squid_krb5_test)
+  AS_IF([test "x$squid_cv_working_krb5" = "xyes"],[
+    AC_DEFINE(HAVE_KRB5,1,[KRB5 support])
+  ],[test "x$squid_cv_working_krb5" = "xno" -a `echo $LIBS | grep -i -c "(-)L"` -gt 0],[
+    AC_MSG_NOTICE([Check Runtime library path !])
+  ])
+])
 
-        return 0;
-}
-]])], [ squid_cv_broken_heimdal_krb5_h=yes ], [ squid_cv_broken_heimdal_krb5_h=no ])
+dnl check how to access the max skew in the krb5 context
+AC_DEFUN([SQUID_CHECK_KRB5_GET_MAX_TIME_SKEW],[
+  AC_CACHE_CHECK([for max_skew in struct krb5_context],squid_cv_max_skew_context,[
+    SQUID_STATE_SAVE(squid_krb5_test)
+    CPPFLAGS="-I${srcdir:-.} $CPPFLAGS"
+    AC_CHECK_LIB(krb5,[krb5_get_max_time_skew],[
+      squid_cv_max_skew_context="yes"],[
+      AC_COMPILE_IFELSE([
+        AC_LANG_PROGRAM([[#include "compat/krb5.h"]],[[krb5_context kc; kc->max_skew = 1;]])
+      ],[squid_cv_max_skew_context=replace],[squid_cv_max_skew_context=no],[squid_cv_max_skew_context=no])
```

Comment:
Maybe I am counting brackets incorrectly, but it looks like this AC_COMPILE_IFELSE() call has a total of four actual parameters instead of the three allowed by [autoconf](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Running-the-Compiler.html):

1. input (`AC_LANG_PROGRAM`)
2. action-if-true (`squid_cv_max_skew_context=replace`)
3. action-if-false (the first `squid_cv_max_skew_context=no`)
4. ??? (the second `squid_cv_max_skew_context=no`)

Reply:
Ah yes. mixup with AC_RUN_IFELSE. Fixed in [arc-autoconf-ng-13](https://github.com/squid-cache/squid/pull/1157/commits/5799bdfcd48d4f448769ee25af39e40529ef7b93) along with one other.